crates/oxc_query/src/vertex.rs

```diff
@@ -0,0 +1,319 @@
+                InterfaceExtendVertex::MemberExpression(_) => "MemberExtend",
+            },
+            Vertex::JSXAttribute(_) => "JSXAttribute",
+            Vertex::JSXElement(jsx) => jsx.type_name(),
+            Vertex::JSXExpressionContainer(_) => "JSXExpressionContainer",
+            Vertex::JSXFragment(_) => "JSXFragment",
+            Vertex::JSXOpeningElement(_) => "JSXOpeningElement",
+            Vertex::JSXSpreadAttribute(_) => "JSXSpreadAttribute",
+            Vertex::JSXSpreadChild(_) => "JSXSpreadChild",
+            Vertex::JSXText(_) => "JSXText",
+            Vertex::ObjectLiteral(_) => "ObjectLiteral",
+            Vertex::PathPart(_) => "PathPart".into(),
+            Vertex::SearchParameter(_) => "SearchParameter",
+            Vertex::Span(_) => "Span",
+            Vertex::SpecificImport(_) => "SpecificImport",
+            Vertex::TypeAnnotation(_) => "TypeAnnotation",
+            Vertex::Type(_) => "Type",
+            Vertex::URL(_) => "URL",
+            Vertex::VariableDeclaration(_) => "VariableDeclaration",
+            Vertex::ReturnStatementAST(_) => "ReturnStatementAST",
+        }
+    }
+
+    pub fn span(&self) -> Span {
+        match &self {
+            Self::AssignmentType(data) => data.span(),
+            Self::ASTNode(data) => data.kind().span(),
+            Self::Class(data) => data.class.span,
+            Self::ClassMethod(data) => data.method.span,
+            Self::ClassProperty(data) => data.property.span,
+            Self::DefaultImport(data) => data.span,
+            Self::Expression(data) => data.span(),
+            Self::Import(data) => data.import.span,
+            Self::Interface(data) => data.interface.span,
+            Self::InterfaceExtend(data) => match **data {
+                InterfaceExtendVertex::Identifier(ident) => ident.span,
+                InterfaceExtendVertex::MemberExpression(membexpr) => (*membexpr).span(),
+            },
+            Self::JSXAttribute(data) => data.span,
+            Self::JSXElement(data) => data.element.span,
+            Self::JSXExpressionContainer(data) => data.span,
+            Self::JSXFragment(data) => data.span,
+            Self::JSXOpeningElement(data) => data.span,
+            Self::JSXSpreadAttribute(data) => data.span,
+            Self::JSXSpreadChild(data) => data.span,
+            Self::JSXText(data) => data.span,
+            Self::ObjectLiteral(data) => data.span,
+            Self::SpecificImport(data) => data.span,
+            Self::TypeAnnotation(data) => data.type_annotation.span,
+            Self::Type(data) => data.span(),
+            Self::URL(_) => unreachable!(),
+            Self::VariableDeclaration(data) => data.variable_declaration.span,
+            Self::ReturnStatementAST(data) => data.return_statement.span,
+            Self::File | Self::PathPart(_) | Self::SearchParameter(_) | Self::Span(_) => {
+                unreachable!()
```

Comment:
It's a good idea to include relevant information in the `unreachable!()` statement to help debug the problem, such as the value of `self` -- was it a `File`, `PathPart`, `SearchParameter`, or `Span`?

Reply:
[cca0c97](https://github.com/web-infra-dev/oxc/pull/599/commits/cca0c9749fe4f5ff9e819a098e2187e8d1b3637a)